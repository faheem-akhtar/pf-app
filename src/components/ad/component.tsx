import { FunctionComponent } from 'react';

import styles from './ad.module.scss';

export const AdComponent: FunctionComponent<{ id: string }> = ({ id }) => <div data-ad className={styles.ad} id={id} />;
