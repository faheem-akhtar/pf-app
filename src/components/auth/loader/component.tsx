import { domClassMerge } from 'helpers/dom/class-merge';

import styles from './auth-loader.module.scss';

export const AuthLoaderComponent = ({
  isEnabled,
  isCentered,
}: {
  isEnabled: boolean;
  isCentered: boolean;
}): JSX.Element | null =>
  isEnabled ? (
    <div className={styles.loader1}>
      <div
        className={domClassMerge(styles['loader1__spinner'], isCentered ? styles['loader1__spinner--centered'] : '')}
      >
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--1'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--2'])} />
        <div className={domClassMerge(styles['loader1__bounce'], styles['loader1__bounce--3'])} />
      </div>
    </div>
  ) : null;
