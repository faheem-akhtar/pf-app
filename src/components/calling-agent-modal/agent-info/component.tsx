import Image from 'next/image';
import { FunctionComponent } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './calling-agent-modal-agent-info.module.scss';
import { CallingAgentModalAgentInfoComponentPropsInterface } from './component-props.interface';

export const CallingAgentModalAgentInfoComponent: FunctionComponent<CallingAgentModalAgentInfoComponentPropsInterface> =
  ({ referenceId, imageSrc, languages, name }) => {
    const { t } = useTranslation();

    return (
      <div className={styles.container}>
        <div className={styles['image-container']}>
          {imageSrc ? (
            <picture>
              <source srcSet={imageSrc} type='image/jpeg' />
              <img className={styles.image} src={imageSrc} alt={name} />
            </picture>
          ) : (
            <Image
              src='/static/images/agent-image-placeholder.jpg'
              width={70}
              height={70}
              objectFit='cover'
              alt='Picture of the agent'
            />
          )}
        </div>
        <div className={styles.content}>
          <p className={styles.name}>{name}</p>
          <p className={styles.description}>
            {languages?.length ? (
              <>
                {`${t('speaks')}: ${languages.join(', ')}`}
                <br />
              </>
            ) : null}
            {t('reference')}: <span className={styles.ref}>{referenceId}</span>
          </p>
        </div>
      </div>
    );
  };
