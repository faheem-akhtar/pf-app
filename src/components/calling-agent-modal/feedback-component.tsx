import React, { useContext } from 'react';
import { useTranslation } from 'helpers/translation/hook';

import { apiReportFetcher } from 'api/report/fetcher';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { CountryCodeEnum } from 'enums/country/code.enum';
import { FiltersContext } from 'components/filters/context';
import { FiltersParametersEnum } from 'enums/filters/parameters.enum';
import { ReportAttributesInterface } from 'types/report/attributes-interface';
import { UserContext } from 'context/user/context';

import styles from './calling-agent-modal.module.scss';
import { userReportCategory } from 'helpers/user/report-category';

const FeedbackAnswerKeysInOrder = ['yes', 'no', 'agent-modal/agent-not-answered', 'agent-modal/not-called'];
const AnswersNeedToBeReported = ['no'];

const reportAttributes: Omit<ReportAttributesInterface, 'reporter_type'> = {
  email: 'report@report.com',
  message: 'Property Not Available - Call Lead Pop-up',
  reason_id: 1,
};

const isCountryAE = process.env.NEXT_PUBLIC_COUNTRY_CODE === CountryCodeEnum.ae;

export const CallingAgentModalFeedbackComponent: React.FunctionComponent<{
  propertyId: string;
  onAnswerClicked: () => void;
}> = ({ onAnswerClicked, propertyId }) => {
  const filtersCtx = useContext(FiltersContext);
  const user = useContext(UserContext);
  const { t } = useTranslation();

  const onClickAnswer = (option: string) => (): void => {
    if (AnswersNeedToBeReported.includes(option)) {
      apiReportFetcher(propertyId, {
        ...reportAttributes,
        email: user?.email || reportAttributes.email,
        reporter_type: isCountryAE ? userReportCategory(filtersCtx.value[FiltersParametersEnum.categoryId]) : '',
      });
    }
    onAnswerClicked();
  };

  return (
    <React.Fragment>
      <div className={styles.feedbackContainer}>
        {FeedbackAnswerKeysInOrder.map((option, index) => (
          <ButtonTemplate
            key={`feedback-${index}`}
            className={styles.option}
            type='button'
            componentType={ButtonComponentTypeEnum.secondary}
            size={ButtonSizeEnum.regular}
            onClick={onClickAnswer(option)}
          >
            {t(option)}
          </ButtonTemplate>
        ))}
      </div>
      <span className={styles.feedbackHint}>{t('helping-improvements')}</span>
    </React.Fragment>
  );
};
