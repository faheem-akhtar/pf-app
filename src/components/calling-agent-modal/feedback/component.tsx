import React, { useContext } from 'react';

import { apiReportFetcher } from 'api/report/fetcher';
import { FiltersContext } from 'components/filters/context';
import { UserContext } from 'context/user/context';
import { FiltersParametersEnum } from 'enums/filters/parameters.enum';
import { PropertyReportReasonEnum } from 'enums/property/report/reason.enum';
import { useTranslation } from 'helpers/translation/hook';
import { userReportCategory } from 'helpers/user/report-category';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { ReportAttributesInterface } from 'types/report/attributes-interface';

import { FEEDBACK_ANSWER_KEYS_IN_ORDER } from './answer-keys-in-order.constant';
import { FEEDBACK_ANSWERS_NEED_TO_BE_REPORTED } from './answers-need-to-be-reported.constant';
import styles from './calling-agent-modal-feedback.module.scss';
import { CallingAgentModalFeedbackComponentPropsInterface } from './component-props.interface';

const reportAttributes: Omit<ReportAttributesInterface, 'reporter_type'> = {
  email: 'report@report.com',
  message: 'Property Not Available - Call Lead Pop-up',
  reason_id: PropertyReportReasonEnum.notAvailable,
};

export const CallingAgentModalFeedbackComponent: React.FunctionComponent<CallingAgentModalFeedbackComponentPropsInterface> =
  ({ onAnswerClicked, propertyId }) => {
    const filtersCtx = useContext(FiltersContext);
    const user = useContext(UserContext);
    const { t } = useTranslation();

    const onClickAnswer = (option: string) => (): void => {
      if (FEEDBACK_ANSWERS_NEED_TO_BE_REPORTED.includes(option)) {
        apiReportFetcher(propertyId, {
          ...reportAttributes,
          email: user?.email || reportAttributes.email,
          reporter_type: userReportCategory(filtersCtx.value[FiltersParametersEnum.categoryId]),
        });
      }
      onAnswerClicked(option);
    };

    return (
      <React.Fragment>
        <div className={styles.feedback}>
          {FEEDBACK_ANSWER_KEYS_IN_ORDER.map((option, index) => (
            <ButtonTemplate
              key={`feedback-${index}`}
              className={styles.option}
              type='button'
              componentType={ButtonComponentTypeEnum.secondary}
              size={ButtonSizeEnum.regular}
              onClick={onClickAnswer(option)}
            >
              {t(option)}
            </ButtonTemplate>
          ))}
        </div>
        <span className={styles.hint}>{t('helping-improvements')}</span>
      </React.Fragment>
    );
  };
