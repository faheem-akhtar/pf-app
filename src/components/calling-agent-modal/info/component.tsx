import React, { Fragment } from 'react';

import { useTranslation } from 'helpers/translation/hook';

import styles from './calling-agent-modal-info.module.scss';
import { CallingAgentModalInfoComponentPropsInterface } from './component-props.interface';

export const CallingAgentModalInfoComponent: React.FunctionComponent<CallingAgentModalInfoComponentPropsInterface> = ({
  referenceId,
  avatar,
  languages,
  name,
}) => {
  const { t } = useTranslation();

  return (
    <div className={styles.container} data-testid='agent-info-component-details'>
      <div className={styles.avatar}>
        <picture className={styles['avatar-image']}>
          <source srcSet={avatar} type='image/jpeg' />
          <img className={styles['avatar-image']} src={avatar} alt={name} />
        </picture>
      </div>
      <div className={styles.text}>
        <span className={styles['text-name']}>{name}</span>
        <p className={styles['text-description']}>
          {languages?.length ? (
            <Fragment>
              {`${t('speaks')}: ${languages.join(', ')}`}
              <br />
            </Fragment>
          ) : null}
          {t('reference')}: <span className={styles['text-ref']}>{referenceId}</span>
        </p>
      </div>
    </div>
  );
};
