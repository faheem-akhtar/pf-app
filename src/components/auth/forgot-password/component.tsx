import React, { useState } from 'react';

import { AuthForgotPasswordFieldEnum } from 'components/auth/forgot-password/field.enum';
import { AuthForgotPasswordPropsInterface } from 'components/auth/forgot-password/props.interface';
import { AuthLoaderComponent } from 'components/auth/loader/component';
import { AuthLoginFieldEnum } from 'components/auth/login/field.enum';
import { AuthResetPasswordService } from 'services/auth/reset-password.service';
import { ButtonComponentTypeEnum } from 'library/button/component-type.enum';
import { ButtonSizeEnum } from 'library/button/size.enum';
import { ButtonTemplate } from 'library/button/template';
import { domClassMerge } from 'helpers/dom/class-merge';
import { ErrorMessageComponent } from 'components/error-message/component';
import { formMakeValidator } from 'components/form/make-validator';
import { GoogleRecaptchaService } from 'services/google/recaptcha.service';
import { InputBaseComponent } from 'library/input/base/component';
import { ReCaptchaComponent } from 'components/re-captcha/component';
import { useTranslation } from 'helpers/translation/hook';
import { validationEmail } from 'helpers/validation/email';
import { validationRequired } from 'helpers/validation/required';

import styles from '../auth.module.scss';

export const AuthForgotPasswordComponent = (props: AuthForgotPasswordPropsInterface): JSX.Element => {
  const captchaService = GoogleRecaptchaService();
  const { t } = useTranslation();
  const [email, setEmail] = useState('');
  const [errors, setErrors] = useState<Partial<Record<AuthLoginFieldEnum, string>>>({});
  const [errorMessage, setErrorMessage] = useState('');
  const [validators] = useState(() => ({
    [AuthLoginFieldEnum.email]: [
      validationRequired(t('validation/empty-email')),
      validationEmail(t('validation/not-valid-email')),
    ],
  }));
  const [isLoading, setIsLoading] = useState(false);

  /**
   * Validate form fields
   */
  const validate = formMakeValidator(errors, setErrors, validators);

  // TODO-FE[CX-499] - Add success message
  return (
    <div data-testid='AuthForgotPasswordComponent'>
      <div className={styles.heading}>{t('auth/forgot-password')}?</div>
      <AuthLoaderComponent isEnabled={isLoading} isCentered={true} />
      {errorMessage && <ErrorMessageComponent message={errorMessage} />}
      <form
        onSubmit={(e): void => {
          e.preventDefault();

          if (
            !validate({
              [AuthForgotPasswordFieldEnum.email]: email,
            })
          ) {
            setIsLoading(true);
            captchaService.execute().then((captcha_token) => {
              AuthResetPasswordService({
                email,
                captcha_token,
              })
                .then((e) => {
                  if (e.ok) {
                    // Reset server error message
                    setErrorMessage('');
                    // Close modal
                    props.onClose();
                    props.onSuccess();
                  } else {
                    setErrorMessage(e.error?.body || `${t('auth/something-wrong')}! ${t('auth/try-later')}`);
                    captchaService.reset();
                  }
                })
                .finally(() => setIsLoading(false));
            });
          }
          return;
        }}
      >
        <div className={styles['input-area']}>
          <InputBaseComponent
            placeholder={t('email')}
            type='email'
            value={email}
            error={!!errors[AuthForgotPasswordFieldEnum.email]}
            onChange={(value): void => {
              validate({
                [AuthForgotPasswordFieldEnum.email]: value,
              });
              setEmail(value);
            }}
            errorText={errors[AuthForgotPasswordFieldEnum.email]}
          />
        </div>
        <div className={styles['input-area']}>
          <ButtonTemplate
            className={styles.button}
            type='submit'
            componentType={ButtonComponentTypeEnum.primary}
            size={ButtonSizeEnum.regular}
          >
            {t('auth/reset-password')}
          </ButtonTemplate>
        </div>
        <ReCaptchaComponent />
      </form>
      <div className={domClassMerge(styles.link, styles['create-account'])} onClick={props.onRegister}>
        {t('auth/not-registered-yet')}?&nbsp;{t('auth/create-account')}
      </div>
    </div>
  );
};
