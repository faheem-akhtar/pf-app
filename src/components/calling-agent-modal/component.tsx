import React, { Fragment, useRef, useState } from 'react';

import { useApiAgent } from 'api/agent/hook';
import { IconThickCrossTemplate } from 'components/icon/thick/cross-template';
import { ModalComponent } from 'components/modal/component';
import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { functionNoop } from 'helpers/function/noop';
import { useTranslation } from 'helpers/translation/hook';

import styles from './calling-agent-modal.module.scss';
import { CallingAgentModalComponentPropsInterface } from './component-props.interface';
import { CallingAgentModalFeedbackComponent } from './feedback/component';
import { CallingAgentModalInfoComponent } from './info/component';
import { callingAgentModalTracker } from './tracker';

export const CallingAgentModalComponent: React.FunctionComponent<CallingAgentModalComponentPropsInterface> = ({
  property,
  referenceId,
  openRef,
  closeRef,
}) => {
  const [modalIsOpened, setModalIsOpened] = useState(false);
  const [currentStep, setCurrentStep] = useState<number>(0);
  const { t } = useTranslation();
  const internalCloseRef = useRef<() => void>(closeRef?.current || functionNoop);
  const propertyId = propertySerpObfuscatedGetId(property);
  const agentDetailsResponse = useApiAgent(propertyId, modalIsOpened);

  const closeModal = (): void => {
    setCurrentStep(0);
    setModalIsOpened(false);
    internalCloseRef.current();
  };
  const onCloseClick = (): void => {
    if (currentStep === 1) {
      closeModal();
    } else {
      setCurrentStep(currentStep + 1);
    }
  };

  const isAgentInfoShown = currentStep === 0;

  return (
    <ModalComponent
      openRef={openRef}
      closeRef={internalCloseRef}
      overlay
      onOpen={(): void => {
        setModalIsOpened(true);
      }}
    >
      <div className={styles.content} data-testid='calling-agent-modal-content'>
        <div className={styles['content-header']}>
          <h1 className={styles['content-text']}>
            {t(isAgentInfoShown ? 'agent-modal/title' : 'agent-modal/property-availability')}
          </h1>
          <button onClick={onCloseClick}>
            <IconThickCrossTemplate class={styles['content-close-icon']} />
          </button>
        </div>
        {agentDetailsResponse.ok && (
          <Fragment>
            {isAgentInfoShown ? (
              <CallingAgentModalInfoComponent
                avatar={agentDetailsResponse.data.imageSrc}
                name={agentDetailsResponse.data.name}
                languages={agentDetailsResponse.data.languages}
                referenceId={referenceId}
              />
            ) : (
              <CallingAgentModalFeedbackComponent
                propertyId={propertyId}
                onAnswerClicked={(answer): void => {
                  callingAgentModalTracker.onAnswerClicked(property, answer);
                  closeModal();
                }}
              />
            )}
          </Fragment>
        )}
      </div>
    </ModalComponent>
  );
};
