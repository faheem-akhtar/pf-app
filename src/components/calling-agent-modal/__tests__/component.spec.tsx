import { act, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { mockModalEnv } from 'mocks/modal-env/mock';
import { mockReactUseSwr } from 'mocks/react/use-swr.mock';
import { mockWindowFetch } from 'mocks/window/fetch.mock';
import { filtersContextPropsStub } from 'stubs/filters/context-props.stub';
import { propertyStub } from 'stubs/property/stub';

import { FiltersContextProvider } from 'components/filters/context-provider';
import { propertySerpObfuscatedGetId } from 'components/property/serp/obfuscated/get/id';
import { propertySerpObfuscatedGetReference } from 'components/property/serp/obfuscated/get/reference';
import { PropertySerpObfuscatedType } from 'components/property/serp/obfuscated/type';
import { PropertyReportReasonEnum } from 'enums/property/report/reason.enum';
import { PropertyReportUserTypeEnum } from 'enums/property/report/user-type.enum';

import { CallingAgentModalComponent } from '../component';
import { CallingAgentModalComponentPropsInterface } from '../component-props.interface';
import * as CallingAgentModalFeedbackComponentModule from '../feedback-component';

describe('CallingAgentModalComponent', () => {
  const openRef = { current: jest.fn() };
  const closeRef = { current: jest.fn() };
  const property: PropertySerpObfuscatedType = propertyStub();
  const propertyId: string = propertySerpObfuscatedGetId(property);

  let props: CallingAgentModalComponentPropsInterface;

  beforeEach(() => {
    mockModalEnv();

    window.dataLayer = [];

    props = {
      property,
      referenceId: propertySerpObfuscatedGetReference(property),
      openRef,
      closeRef,
    };
  });

  it('should render without throwing any errors', () => {
    mockReactUseSwr('en-property-search/agent-GET-{"propertyId":"198023"}', {
      ok: false,
    });
    render(<CallingAgentModalComponent {...props} />);
    act(openRef.current);

    expect(screen.getByTestId('calling-agent-modal-content')).toMatchSnapshot();
  });

  describe('when agent model information is loaded', () => {
    beforeEach(() => {
      mockReactUseSwr('en-property-search/agent-GET-{"propertyId":"198023"}', {
        ok: true,
        data: { name: 'lorem', languages: ['a', 'b'], imageSrc: 'image' },
      });

      render(<CallingAgentModalComponent {...props} />);
      act(openRef.current);
    });

    it('should update title when close is clicked first time', () => {
      userEvent.click(screen.getByRole('button', { name: /cross/i }));

      expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('agent-modal/property-availability');
    });

    it('should close modal when close is clicked second time', () => {
      const closeButton = screen.getByRole('button', { name: /cross/i });
      userEvent.dblClick(closeButton);

      expect(closeButton).not.toBeInTheDocument();
    });

    test('if agent info is shown properly', () => {
      expect(screen.getByTestId('agent-info-component-details')).toMatchSnapshot();
    });

    it('should render feedback component when agent data exists and close is clicked first time', () => {
      userEvent.click(screen.getByRole('button', { name: /cross/i }));

      expect(screen.getByText('agent-modal/agent-not-answered')).toBeInTheDocument();
    });
  });

  describe('FeedbackComponent', () => {
    it('should call onAnswerClicked callback when a feedback button is clicked', async () => {
      mockReactUseSwr('en-property-search/agent-GET-{"propertyId":"198023"}', {
        ok: true,
        data: { name: 'lorem', languages: ['a', 'b'], imageSrc: 'image' },
      });
      jest
        .spyOn(CallingAgentModalFeedbackComponentModule, 'CallingAgentModalFeedbackComponent')
        .mockImplementationOnce(({ onAnswerClicked }) => (
          <button data-testid='feedback-component-button' onClick={(): void => onAnswerClicked('yes')} />
        ));

      render(<CallingAgentModalComponent {...props} />);

      act(openRef.current);

      const closeButton = screen.getByRole('button', { name: /cross/i });

      expect(closeButton).toBeInTheDocument();

      userEvent.click(closeButton);

      userEvent.click(screen.getByTestId('feedback-component-button'));

      expect(closeButton).not.toBeInTheDocument();

      expect(window.dataLayer).toEqual([
        {
          event: 'customEvent',
          eventCategory: 'Post Lead Survey',
          eventAction: 'propertyAvailable',
          eventLabel: 'offering-type - agentId - broker',
        },
      ]);
    });

    it('should make a request if clicked answer is no', async () => {
      const fetchMock = mockWindowFetch();
      render(
        <FiltersContextProvider {...filtersContextPropsStub()}>
          <CallingAgentModalFeedbackComponentModule.CallingAgentModalFeedbackComponent
            onAnswerClicked={jest.fn()}
            propertyId={propertyId}
          />
        </FiltersContextProvider>
      );
      const answerButton = screen.getByText('no');
      userEvent.click(answerButton);

      await waitFor(() =>
        expect(fetchMock).toHaveBeenCalledWith(
          expect.anything(),
          expect.objectContaining({
            body: JSON.stringify({
              data: {
                type: 'property_report',
                attributes: {
                  email: 'report@report.com',
                  message: 'Property Not Available - Call Lead Pop-up',
                  reason_id: PropertyReportReasonEnum.notAvailable,
                  reporter_type: PropertyReportUserTypeEnum.renter,
                },
              },
            }),
          })
        )
      );
    });
  });
});
