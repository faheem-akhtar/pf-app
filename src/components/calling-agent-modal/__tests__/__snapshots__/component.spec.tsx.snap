// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallingAgentModalComponent should render without throwing any errors 1`] = `
<div
  class="content"
  data-testid="calling-agent-modal-content"
>
  <div
    class="header"
  >
    <h1
      class="text"
    >
      agent-modal/title
    </h1>
    <button>
      <svg
        class="closeIcon"
        viewBox="7 7 10 10"
      >
        <title>
          Cross
        </title>
        <path
          clip-rule="evenodd"
          d="M8.34408 7.17475C8.01717 6.92057 7.54448 6.94367 7.24408 7.24408C6.91864 7.56951 6.91864 8.09715 7.24408 8.42259L10.8215 12L7.24408 15.5774C6.91864 15.9028 6.91864 16.4305 7.24408 16.7559C7.54448 17.0563 8.01717 17.0794 8.34408 16.8252L8.42259 16.7559L12 13.1785L15.5774 16.7559L15.6559 16.8252C15.9828 17.0794 16.4555 17.0563 16.7559 16.7559C17.0814 16.4305 17.0814 15.9028 16.7559 15.5774L13.1785 12L16.7559 8.42259C17.0814 8.09715 17.0814 7.56951 16.7559 7.24408C16.4555 6.94367 15.9828 6.92057 15.6559 7.17475L15.5774 7.24408L12 10.8215L8.42259 7.24408L8.34408 7.17475Z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`CallingAgentModalComponent when agent model information is loaded if agent info is shown properly 1`] = `
<div
  class="details"
  data-testid="agent-info-component-details"
>
  <div
    class="avatarContainer"
  >
    <picture
      class="avatar"
    >
      <source
        srcset="image"
        type="image/jpeg"
      />
      <img
        class="avatar"
        src="image"
      />
    </picture>
  </div>
  <div
    class="info"
  >
    <span
      class="name text"
    >
      lorem
    </span>
    <p
      class="description"
    >
      speaks: a, b
      <br />
      reference
      : 
      <span
        class="text"
      >
        123
      </span>
    </p>
  </div>
</div>
`;
